syft/core/frameworks/torch/tensor.py

```diff
@@ -529,6 +531,79 @@ def abs(self):
         return torch.abs(self)
 
 
+class _MPCTensor(_SyftTensor):
+    """
+    Example of a custom overloaded _SyftTensor
+
+    Role:
+    Converts all add operations into sub/minus ones.
+    """
+
+    def __init__(self, shares, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.shares = shares # shares is a _GeneralizedPointerTensor
+
+
+    # The table of command you want to replace
+    substitution_table = {
+        'torch.add': 'torch.add',
+        'torch.mul': 'torch.mul',
+    }
+
+    class overload_functions:
+        """
+        Put here the functions you want to overload
+        Beware of recursion errors.
+        """
+
+        @staticmethod
+        def get(attr):
+            attr = attr.split('.')[-1]
+            return getattr(sy._MPCTensor.overload_functions, attr)
+
+    # Put here all the methods you want to overload
+
+      def add(self, other):
+        # gp_ stands for GeneralizedPointer
+        gp_response = spdz.spdz_add(self.shares, other.shares)
+        response = _MPCTensor(gp_response
+        # response.shares = shares
+        return response
+
+    def mul(self, other):
+        gp_response = spdz.spdz_mul(self.shares, other.shares)
+        response = _MPCTensor(gp_response)
+        return response
+       
+    def send(self, workers):
+        self.n_workers = len(workers)
+        self.shares = self.share(self.var, self.n_workers)
+        self.workers = workers
+        for share, worker in zip(self.shares, self.workers):
+            share.send(worker)
+
+    def get(self):
+        shares = []
+        for share in self.shares:
+            shares.append(share.get())
+        var = spdz.reconstruct(shares)
+        self.var = var
+        return var
+
+class _GeneralizedPointerTensor(_SyftTensor):
+
+    def __init__(self, pointer_tensor_dict, parent, torch_type, id = None, owner=None, skip_register=False):
+         super().__init__(child=None, parent=parent, torch_type=torch_type, owner=owner, id=id,
+                         skip_register=skip_register)
+         self.pointer_tensor_dict = pointer_tensor_dict
+    
+    @classmethod
+    def handle_call(cls, syft_command, owner):
+        commands = torch_utils.split_to_pointer_commands(syft_command)
+        result_dict = {worker_id:_PointerTensor.handle_call(commands[worker_id], owner) for worker_id in commands}
+        #TODO: @trask @theo could you take a look at this if you have better ideas on how to get these parameters
+        return _GeneralizedPointerTensor(result_dict, None, None, id=None, owner=owner, skip_register=False)
```

Comment:
The attributes for the new _GeneralizedPointerTensor should be either created or pulled from result_dict. 

parent - Normally, the "parent" would come from the high level torch object wrapping the result that's returned to you, but since you'll have multiple, I suppose the only thing to do is to pick one at random. 

torch_type - you should be able to get this from any of the answers in result_dict, since they should all be identical (optional: a check to ensure that they are in fact identical)

id - this should be initialized by _GeneralizedPointerTensor automatically when it calls super().__init__ on line 596. Might be good to check that it is though.

skip_register - i think setting this to False makes sense... (which you have done)

Reply:
@LaRiffle feel free to comment

Reply:
Just one point regarding the "parent", the usual case would be, as far as I can imagine
```
x = sy.FloatTensor([..])
x = sy._MPCTensor().on(x)
x.send([alice, bob])
```
And _MPCTensor would use under the hood the _GeneralizedPointerTensor. Since the parent of _MPCTensor is clear, _GeneralizedPointerTensor should inherit the same one I think.

Reply:
Agreed - although I think that in the specific line i was referring to... a method is being called generating a new _GPT class which doesn't necessarily have a parent yet.